Cpp/Library/Processor/Processor.hh

```diff
@@ -39,7 +39,7 @@ namespace Nymph
         public:
             ConfigException() noexcept;
             ConfigException( const std::string& a_filename, int a_line ) noexcept;
-            ConfigException( const scarab::param_node& node ) noexcept;
+            ConfigException( const scarab::param& config ) noexcept;
```

Comment:
I assume the intention of this change is to allow its use with other derived types of the `param` class? What would be a use case and why does the `Processor` class still require the specific `param_node` in its configuration?

Reply:
The motivation for this change was to use it with `param_array` in `ProcessorToolbox::ConfigureProcessors()`, `ConfigureConnections()`, and `ConfigureRunQueue()`, all of which get configured with arrays.  What really needs to happen is that `ConfigureException` gets moved out of `Processor.hh/cc` and into its own files.  I'll add this as a future refactoring task, along with the points below about `ProcessorToolbox`.